Exception Handling

1.	raise exceptions.
2.	catch exceptions.
3.	error message.
4a.	  return from raising function immediately - but continue in raising function caller.
4b.	  exit.
4c.	  return to function which sets 'catch'
_ <above seem doable>
5.	block area for catch as per try...catch.
6.	nested traps/exceptions - go back to catching 'previous' exception afterer 'block' area.
7.	named exception types - rethrow if not correct exception.


Notes:

documentation approach.

thisfn()
{
	do stuff that sets up for docco.
	end processing of that.
	fn.body-starts or something raises exception - exit fn.body ... how to exit thisfn?
}
